app/routes.py

```diff
@@ -1,2 +1,325 @@
+
+@customer_bp.route("", methods=["POST"], strict_slashes=False)
+def create_new_customer():
+    request_body = request.get_json()
+
+    if not request_body:
+            return make_response({"details": "Invalid data"}, 400)
+    
+    
+    if not request_body.get("name"): 
+        return make_response({"details": "Please enter a value for Name"}, 400)
+    elif not request_body.get("postal_code"):
+        return make_response({"details": "Please enter a value for postal code"}, 400)
+    elif not request_body.get("phone"):
+        return make_response({"details": "Please enter a value for phone"}, 400)
+
+    new_customer = Customer.from_json(request_body)
+
+    db.session.add(new_customer)
+    db.session.commit()
+
+    return make_response({"id": f"{new_customer.id}"}, 201)
+
+@customer_bp.route("", methods=["GET"], strict_slashes=False)
+def get_all_customers():
+    customers = Customer.query.all()
+
+    if customers is None:
+        return make_response([], 404)
+
+    customers_response = [customer.make_json() for customer in customers]
+
+    return make_response(jsonify(customers_response),200)
+
+
+@customer_bp.route("/<customer_id>", methods=["GET"], strict_slashes=False)
+def get_customer_by_id(customer_id):
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return make_response({"details": f"Customer {customer_id} not found"}, 404)
+    
+    return make_response(customer.make_json(), 200)
+
+@customer_bp.route("/<customer_id>", methods=["PUT"], strict_slashes=False)
+def update_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return make_response({"details": f"Customer {customer_id} not found"}, 404)
+    
+    request_data = request.get_json()
+    if not request_data.get("name") or type(request_data.get("name")) != str:
+        return make_response({"details": "Name field is missing or of an invalid data type"}, 400)
+    elif not request_data.get("postal_code") or type(request_data.get("postal_code")) != str:
+        return make_response({"details": "Postal_code field is missing or of an invalid data type"}, 400)
+    elif not request_data.get("phone") or type(request_data.get("phone")) != str:
+        return make_response({"details": "Phone number field is missing or of an invalid data type"}, 400)
+    
+    customer.name = request_data["name"]
+    customer.postal_code = request_data["postal_code"]
+    customer.phone = request_data["phone"]
+
+    db.session.commit()
+    return make_response(customer.make_json(), 200)
+
+@customer_bp.route("/<customer_id>", methods=["DELETE"], strict_slashes=False)
+def delete_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+
+    if customer is None:
+        return make_response({"details": f"Customer {customer_id} not found"}, 404) 
+
+    db.session.delete(customer)
+    db.session.commit()
+
+    return make_response({"id": f"{customer.id}"}, 200)   
```

Comment:
Since the id is universally an integer, I would make it an int.

```suggestion
    return make_response({ "id": customer.id }, 200)   
```